plugins/org.eclipse.fordiac.ide.typemanagement/src/org/eclipse/fordiac/ide/typemanagement/refactoring/SafeStructDeletionChange.java

```diff
@@ -244,45 +168,25 @@ public EnumSet<ChangeState> getDefaultSelection() {
 
 	}
 
-	public static class DeleteUntypedSubappPinsChange extends CompositeChange {
-
-		final SubApp subapp;
-		final StructuredType struct;
+	public static class DeleteUntypedSubappPinChange extends CompositeChange {
+		VarDeclaration varDecl;
 
-		public DeleteUntypedSubappPinsChange(final SubApp subapp, final StructuredType struct) {
+		public DeleteUntypedSubappPinChange(final VarDeclaration varDecl) {
 			super(MessageFormat.format(Messages.DeleteFBTypeParticipant_Change_DeleteSubappPins,
-					subapp.getQualifiedName()));
-			this.subapp = subapp;
-			this.struct = struct;
-			// could be problematic because as far as i know there is no "update FBs" dialog
-			// if you save a subapp that is contained inside a cfb/typed subapp
-			if (!hasOpenEditor(subapp)) {
-				addUpdateChanges();
-			}
+					getSubappName(varDecl)));
+			this.varDecl = varDecl;
 		}
 
-		// triggers the update chain if the subapp is contained inside a typed network
-		private void addUpdateChanges() {
-			// CFB cannot contain a subapp -> only a typed subapp can
-			if (EcoreUtil.getRootContainer(subapp) instanceof final SubAppType type) {
-				SafeFBTypeDeletionChange.addUpdateChanges(this, type, false);
-			}
+		private static String getSubappName(final VarDeclaration varDecl) {
+			return ((SubApp) varDecl.eContainer().eContainer()).getQualifiedName();
 		}
 
 		@Override
 		public Change perform(final IProgressMonitor pm) throws CoreException {
-			final CompoundCommand cmd = new CompoundCommand();
-			final String structFullName = PackageNameHelper.getFullTypeName(struct);
-			// @formatter:off
-			subapp.getInterface().getAllInterfaceElements().stream()
-				.filter(ie -> PackageNameHelper.getFullTypeName(ie.getType()).equalsIgnoreCase(structFullName))
-				.map(DeleteSubAppInterfaceElementCommand::new)
-				.forEach(cmd::add);
-			// @formatter:on
-			ChangeExecutionHelper.executeChange(cmd, subapp, pm);
+			final Command cmd = new DeleteSubAppInterfaceElementCommand(varDecl);
+			ChangeExecutionHelper.executeChange(cmd, varDecl, pm);
```

Comment:
`ChangeExeuctionHelper.executeChange` is deprecated. Please use `AbstractLiveSearchContext.executeAndSave`

Reply:
thanks!